docs/guidelines_iac.md

```diff
@@ -0,0 +1,108 @@
+# IaC Configuration Guideline
+
+This document outlines the standard Terraform configuration and provides guidelines.
+
+## Infrastructure as Code
+
+The `repository uses `[Terraform](https://www.terraform.io/) for managing infrastructure as code (IaC). Terraform provides a predictable,
+safe, and efficient approach to creating, modifying, and managing infrastructure. IaC simplifies:
+
+- Reproducibility of infrastructure through a single source of truth.
+- Pre-validation of infrastructure changes before deployment.
+- Infrastructure change tracking and history.
+- Easy rollback and improved documentation through code integration.
+
+## IaC Configuration Structure
+
+Our configuration aligns with [Google Terraform best practices](https://cloud.google.com/docs/terraform/best-practices-for-terraform)
+and [Hashicorp module development guidelines](https://developer.hashicorp.com/terraform/language/modules/develop), facilitating module reuse
+and streamlined testing.
+
+Terraform configurations must reside within the `terraform` directory, structured and named per application or use case. Grouping by
+resource type is not permitted.
+
+For better readbility and maintenance in config definitions must always be preffered over cli parameters or environment variables.
```

Comment:
```suggestion
For better config readability and maintenance, definitions must always be preferred over cli parameters or environment variables.
```

Reply:
A phrase `in config definitions` is important, as we want to stress that we prefer data in config rather than in cli parameters. It's not about `definitions` alone as you can't have it in cli parameters. However, certain data you can define in config or as cli parameter.

Reply:
Oh, ok. Now I get it. I updated the comment accordingly. Thanks!